### DISCUSSION TITLE
FileChooser Widget

### DISCUSSION THREAD

**Discussion by lovetox** [Textualize/textual#1770] [Q&A] [answered]
Hi,

Did anybody implement a FileChooser Widget? Seems like a very common thing one would need for such console applications.

Thanks

**Comment by davep** [comment, marked as answer]
There is a builtin [`DirectoryTree` widget](https://textual.textualize.io/widgets/directory_tree/) which can serve that purpose (I've used it for that myself [in one of my own applications](https://www.youtube.com/watch?v=wfIttBtlun0)). If on the other hand you mean a more traditional file open/save dialog sort of widget, there isn't anything in Textual for that right now and I've not seen any examples of third party widgets yet (that isn't to say they don't exist, just that I've not personally seen any).
